tests/Dumplie/Test/Integration/Application/InMemory/Metadata/MetadataTest.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+namespace Dumplie\Test\Integration\Application\InMemory\Metadata;
+
+use Dumplie\Application\Metadata\Hydrator\DefaultHydrator;
+use Dumplie\Application\Metadata\Metadata;
+use Dumplie\Application\Metadata\MetadataAccessRegistry;
+use Dumplie\Application\Metadata\MetadataId;
+use Dumplie\Application\Metadata\Schema;
+use Dumplie\Application\Metadata\Storage;
+use Dumplie\Infrastructure\InMemory\Metadata\InMemoryStorage;
+
+class MetadataTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @var Storage
+     */
+    private $storage;
+
+    /**
+     * @var MetadataAccessRegistry
+     */
+    private $registry;
+
+    public function setUp()
+    {
+        $hydrator = new DefaultHydrator();
+
+        $schema = new Schema();
+
+        $schema->add(new Schema\TypeSchema("product", [
+            "sku" => new Schema\Field\TextField(),
+            "name" => new Schema\Field\TextField(),
+            "brand" => new Schema\Field\TextField("Dumplie")
+        ]));
+
+        $this->storage = new InMemoryStorage();
+        $this->registry = new MetadataAccessRegistry($this->storage, $schema, $hydrator);
+    }
+
+    /**
+     * @expectedException \Dumplie\Application\Exception\Metadata\NotFoundException
```

Comment:
Maybe instead annotation lets use `$this->expectException(Dumplie\Application\Exception\Metadata\NotFoundException::class);` ?